src/interpreter/mod.rs

```diff
@@ -734,16 +734,12 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
             ty::TyRef(_, ty::TypeAndMut { ty, .. }) |
             ty::TyRawPtr(ty::TypeAndMut { ty, .. }) |
             ty::TyBox(ty) => {
-                if self.type_is_sized(ty) {
-                    assert_eq!(field_index, 0);
-                    Ok(self.tcx.mk_imm_ptr(self.tcx.types.never))
-                } else {
-                    match (field_index, &ty.sty) {
-                        (1, &ty::TySlice(_)) => Ok(self.tcx.types.usize),
-                        (1, &ty::TyTrait(_)) |
-                        (0, _) => Ok(self.tcx.mk_imm_ptr(self.tcx.types.never)),
-                        _ => bug!("invalid fat pointee type"),
-                    }
+                assert!(!self.type_is_sized(ty));
+                match (field_index, &self.tcx.struct_tail(ty).sty) {
+                    (1, &ty::TySlice(_)) => Ok(self.tcx.types.usize),
```

Comment:
> It's probably okay to keep it?

I'd prefer to stick to cases we know can happen (`field_index == 0`, or more specifically the constant that says which fat pointer field is the thin pointer) and assert that nothing else happens. Since we can't use length fields, I'd remove any code that handles those cases. Vtable pointers seem plausible, but do we even use those?